### PR TITLE
Add test cases for modulo (%) operator

### PR DESCRIPTION
**Pull request by sasha-gil**
## Summary
This PR adds comprehensive test cases for the modulo (`%`) operator across the nanvm test suite. Since the `%` operator is not yet implemented in `nanvm-lib`, all Rust test cases are marked as TODO comments while JavaScript proof cases are fully functional.

## Key Changes

- **Added `rem()` test function** in `nanvm-lib/tests/test/generated.rs` with 36 test cases covering:
  - Type coercion (null, undefined, booleans, strings, arrays, objects, functions)
  - Basic numeric operations (zero, positive, negative numbers)
  - Edge cases (infinity, NaN, negative zero)
  - Fractional numbers
  - BigInt operations and mixed type errors

- **Added `remCases` test data** in `fjs/nanvm/module.f.mjs` with detailed documentation explaining:
  - Modulo semantics (result sign follows dividend, not divisor)
  - Numeric vs BigInt behavior differences
  - All 36 test cases with expected results and `rust: remNotImplemented` reason

- **Updated test infrastructure** in `fjs/nanvm/rust/module.f.mjs`:
  - Added `'%': 'rem'` mapping to `rustName`
  - Added modulo operation to `op2Rust` object (generates commented syntax)
  - Added documentation explaining why Rust cases are commented out

- **Updated proof implementation** in `fjs/nanvm/proof.f.mjs`:
  - Added `'%': (a, b) => a % b` to `op2Js` for JavaScript execution

- **Updated test group count** in module documentation from 5 to 7 groups

## Implementation Details

All Rust test cases are wrapped in comments with `// TODO: \`%\` is not implemented in nanvm-lib yet` markers. This allows the test infrastructure to:
- Keep test cases documented and ready for when the operator is implemented
- Run JavaScript proofs to validate expected behavior
- Automatically enable Rust tests once the operator is implemented by removing the `rust` reason field

https://claude.ai/code/session_01EtpceTFc1653SDSG7Rsggt

### CODE REVIEW

**Comment by sergey-shandar** [comment]
@codex review

**Review by sergey-shandar — APPROVED** [review]

File: fjs/nanvm/module.f.mjs [hunk 1]
```diff
@@ -377,6 +377,80 @@ const subCases = [
...
+ */
+const remNotImplemented = '`%` is not implemented in nanvm-lib yet'
+
+/**
+ * `%` coerces both operands with `ToNumeric` like `*`, but is neither
+ * commutative nor symmetric between mixed sign operands: the result's sign
+ * follows the dividend (left operand), not the divisor. Number `%` never
+ * throws — division by zero, and either operand infinite or `NaN`, all
+ * produce `NaN` — while BigInt `%` throws on a zero divisor the same way
+ * `nanvm-lib` throws on mixed number/bigint operands.
```
[the remark this answers is not in the document]
Comment (sasha-gil) [reply fjs/nanvm/module.f.mjs]: Good catch — fixed in eb959f1: reworded to state the actual NaN conditions (zero divisor, infinite dividend, either operand NaN) and call out explicitly that a finite dividend over an infinite divisor returns the dividend unchanged.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: nanvm-lib/tests/test/generated.rs [hunk 2]
```diff
@@ -227,6 +227,86 @@ fn add<A: IVm>() {
     check_throws::<A>("bigintPlusNumber", bigint_any(1) + (1f64).to_any());
 }
 
+#[rustfmt::skip]
+fn rem<A: IVm>() {
+    // TODO: `%` is not implemented in nanvm-lib yet
+    // check::<A>("nullModThree", Nullish::Null.to_any() % (3f64).to_any(), (0f64).to_any());
+    // TODO: `%` is not implemented in nanvm-lib yet
+    // check::<A>("undefinedModThree", Nullish::Undefined.to_any() % (3f64).to_any(), (f64::NAN).to_any());
```
[the remark this answers is not in the document]
Comment (sasha-gil) [reply nanvm-lib/tests/test/generated.rs]: Fixed in eb959f1: `emit` in `fjs/nanvm/rust/module.f.mjs` now prints the reason and statement on one `// TODO: ...` line instead of two, and `generated.rs` was regenerated.

---
_Generated by [Claude Code](https://claude.ai/code)_
